static-assets/components/cstudio-common/common-api.js

```diff
@@ -6258,20 +6257,20 @@ var nodeOpen = false,
                 payload: [
                   {
                     type: 'DISPATCH_DOM_EVENT',
-                    payload: { id: onCancel }
+                    payload: { id: confirmDialogEvent, type: 'cancel' }
                   },
                   { type: 'CONFIRM_DIALOG_CLOSED' }
                 ]
               }
             }
           });
 
-          unsubscribe = CrafterCMSNext.createLegacyCallbackListener(onOk, () => {
-            callback();
-            cancelUnsubscribe();
-          });
-          cancelUnsubscribe = CrafterCMSNext.createLegacyCallbackListener(onCancel, () => {
-            unsubscribe();
+          unsubscribe = CrafterCMSNext.createLegacyCallbackListener(confirmDialogEvent, ({ type }) => {
```

Comment:
We could/should replace this with utils.dom.createCustomEventBlahblah, as long as keeps working the same externally, of course.